src/tiler/src/util/layer-filter.js

```
@@ -14,15 +14,91 @@ const parsePromise = xmlString => new Promise((resolve, reject) => {
         else resolve(result)
     })
 })
+/**
+ * Structures sql based on existing query and json options
+ */
+const structureQuery = (existingQuery, currentLayer) => {
+    // Base of the query, wrapping pre-existing query
+    let query = `SELECT * FROM ${existingQuery} WHERE`
+
+    // Add each filter to the query string
+    currentLayer.filters.forEach((filterObj, i) => {
+        // Determine comparison mode (if more than one parameter)
+        let mode = ''
+        if (i > 0) {
+            if (currentLayer.mode) {
+                switch (currentLayer.mode.toUpperCase()) {
+                case 'OR':
+                    mode = 'OR'
+                    break
+                default:
+                    mode = 'AND'
+                    break
+                }
+            } else {
+                mode = 'AND'
+            }
+        }
+
+        // Determine comparison operator
+        let op = '='
+        if (filterObj.op) {
+            switch (filterObj.op.toUpperCase()) {
+            case 'LIKE':
+            case 'IS':
+            case 'IS NOT':
+            case 'IS DISTINCT FROM':
+            case 'IS NOT DISTINCT FROM':
+                op = ` ${filterObj.op.toUpperCase()} `
+                break
+            case '<':
+            case '>':
+            case '<=':
+            case '>=':
+            case '<>':
+            case '!=':
+                /* eslint-disable-next-line prefer-destructuring */
+                op = filterObj.op
+                break
+            default:
+                op = '='
+                break
+            }
+        }
```

It's not immediately clear that these switch blocks aren't really transforming the value (except maybe the case) to a sql-safe value, but it's really more of a white list.  Is the functionality preserved if you do something similar to:

```js
const whitelist = ['LIKE', 'IS', 'IS NOT', 'etc...'];
const modes = ['OR', 'AND'];

const op = whitelist.includes(filterObj.op.toUpperCase()) ? filterObj.op : '=';
const mode = modes.includes(filterObj.mode.toUpperCase()) ? filterOjb.mode : 'AND';
```

---

After reviewing the tests, I'd amend my above sample to throw an exception (invalid syntax) if it failed the whitelist test, instead of defaulting to `=/AND`

---

Those are good ideas, and definitely my intent with that bit.